arangod/RocksDBEngine/RocksDBHotBackup.cpp

```diff
@@ -520,6 +523,34 @@ void RocksDBHotBackupCreate::executeCreate() {
       retVal = TRI_RenameFile(dirPathTemp.c_str(), dirPathFinal.c_str(), &systemError, &errors);
       _success = (TRI_ERROR_NO_ERROR == retVal);
     } // if
+
+    // write encrypted agency dump if available
+    if (_success && !_agencyDump.isNone()) {
+      std::string agencyDumpFileName = dirPathFinal + TRI_DIR_SEPARATOR_CHAR + "agency.json";
+
+#ifdef USE_ENTERPRISE
+      auto encryptFeature = application_features::ApplicationServer::lookupFeature<EncryptionFeature>("Encryption");
+
+      if (encryptFeature != nullptr) {
+        int fd = TRI_TRACKED_CREATE_FILE(agencyDumpFileName.c_str(), O_WRONLY | O_CREAT | O_TRUNC | TRI_O_CLOEXEC,
+                                     S_IRUSR | S_IWUSR | S_IRGRP);
+        if (fd != -1) {
+          TRI_DEFER(TRI_TRACKED_CLOSE_FILE(fd));
+
+          std::string json = _agencyDump.toJson();
+          auto context = encryptFeature->beginEncryption(fd);
+
+          _success = encryptFeature->writeData(*context.get(), json.c_str(), json.size());
+        } else {
+          _success = false;
+        }
+      } else {
+        basics::FileUtils::spit(agencyDumpFileName, _agencyDump.toJson(), true);
+      }
+#else
+      basics::FileUtils::spit(agencyDumpFileName, _agencyDump.toJson(), true);
+#endif
+    }
```

Comment:
should we consider calculating SHA256 and appending it to file name? (renaming file after rereading to get sha)  I am not requiring, just asking since a corrupt encrypted agency file might get nasty.

Reply:
Not a bad idea. Or at least create an additional `.sha` file

Reply:
hash of encrypted or unencrypted agency?